packages/gatsby/src/redux/run-fast-filters.ts

```diff
@@ -1,45 +1,60 @@
-// @flow
-
-const { prepareRegex } = require(`../utils/prepare-regex`)
+import { IGatsbyNode } from "./types"
+import { GatsbyGraphQLType } from "../../index"
+import { prepareRegex } from "../utils/prepare-regex"
 const { makeRe } = require(`micromatch`)
 import { getValueAt } from "../utils/get-value-at"
 import _ from "lodash"
-const {
+import {
+  DbQuery,
+  IDbQueryQuery,
+  IDbQueryElemMatch,
   objectToDottedField,
   createDbQueriesFromObject,
   prefixResolvedFields,
-} = require(`../db/common/query`)
-const {
+} from "../db/common/query"
+import {
+  FilterOp,
+  FilterCacheKey,
+  FiltersCache,
+  FilterValueNullable,
   ensureEmptyFilterCache,
   ensureIndexByQuery,
   ensureIndexByElemMatch,
   getNodesFromCacheByValue,
   intersectNodesByCounter,
-} = require(`./nodes`)
+  IFilterCache,
+} from "./nodes"
+import { IGraphQLRunnerStats } from "../query/types"
+
+interface IInputQuery {
```

Comment:
Could you add a comment here explaining why you can't use `Record`, in case someone tries to refactor it